docs/enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,134 @@
+
+### Implementation Details/Notes/Constraints
+
+Our validations will need to be updated as follows:
+- A host can only be assigned the arbiter role if the cluster is TNA.
+- TNA clusters must have at least 1 arbiter node.
+- TNA clusters can have 2-5 control plane nodes.
+- TNA clusters' platform must be baremetal.
+- TNA clusters are expected to be TP stating from OCP 4.19.
+
+We need to add arbiter's hardware requirements in:
+- The HW_VALIDATOR_REQUIREMENTS environment variable (by default we'll take
+  the worker's requirements to avoid issues mid-development).
+- The HostTypeHardwareRequirementsWrapper structs used by the operators.
+
+We will auto-assign the arbiter role to a host if the following are true:
+- The host's cluster is TNA.
+- The host's cluster has no arbiter nodes.
+- The host has the minimum hardware requirements for arbiter nodes.
+
+Outside of assisted-service, we also need to do the following:
+- After we update the API in assisted-services, we need to update the vendor
+  directory in assisted-installer-agent.
+- We need to update the function waitForMinMasterNodes in assisted-installer
+  to set the min master nodes to be the minimum of 2 (the current value)
+  and number of control planes (read it from the install-config) minus 1.
+
+Until we have this feature fully implemented, we should have an environment
+variable to decide if we allow TNA clusters (default value is false).
+
+### Risks and Mitigations
+
+The main risk in this enhancement is that because we are treating one of the
+master nodes in a 3 node cluster as an arbiter, we are explicitly evicting
+processes that would otherwise be a normal supported upstream configuration
+such as a compact cluster. We run the risk of new components being critical to
+HA not containing the proper tolerations for running on the arbiter node. One
+of the mitigations we can take against that is to make sure we are testing
+installs.
+
+A couple of risks we run is customers using arbiter nodes with improper disk
+speeds below that recommended for etcd, or a bigger problem being network
+latency. Since etcd is sensitive to latency between members, we should
+validate that arbiter nodes meet minimum requirements for ETCD to function
+properly, in disk and network speeds.
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### UI Impact
+
+The main change is that currently the UI relies on the user specifying the
+number of control plane in order to set the high_availability_mode, but TNA
+cluster can have 2-5 control planes so we should ask for it explicitly and
+validate it with the number of control planes.
+
+### Test Plan
+
+- Add an e2e test that will create a TNA cluster.
+- Add unit tests and subsystem tests for arbiter nodes and TNA clusters.
+
+## Drawbacks
+
+A few drawbacks we have is that we will be creating a new variant of OpenShift
+that implements a new unique way of doing HA for kubernetes. This does mean
```

Comment:
This isn't a drawback of our implementation. This sounds like it's talking more about TNA generally.

This enhancement should focus on the changes in assisted installer rather than TNA as an OpenShift topology.